src/hypothesis/strategies.py

```diff
@@ -173,16 +173,26 @@ def one_of(*args):
         except TypeError:
             pass
 
+    # If we have another one_of() in the list of arguments, pick out
+    # its underlying strategies
+    from hypothesis.searchstrategy.strategies import OneOfStrategy
+    strategies = []
```

Comment:
`strategies` is probably a terrible name there, because we haven’t called `check_strategy()` on these objects yet.